src/software/sensor_fusion/ssl_protobuf_reader.h

```diff
@@ -21,23 +22,24 @@ class SSLProtobufReader
 {
    public:
     /**
-     * Creates a new SSLProtobufReader for data input and filtering
+     * Creates a new SSLProtobufReader for reading and processing protobufs
      */
     explicit SSLProtobufReader();
 
+    virtual ~SSLProtobufReader() = default;
+
     /**
-     * Returns a new Field object containing the most up to date state of the field given
-     * the new GeometryData information
+     * Returns a Field object given geometry data from a protobuf packet
      *
      * @param geometry_packet The SSL_GeometryData packet containing new field data
      *
-     * @return a Field object containing the most up to date state of the field given the
-     * new GeometryData information
+     * @return A Field object representing the field specified with the provided geometry
+     *      If packet_geometry is not a valid packet, then will return std::nullopt
      */
-    Field getFieldData(const SSL_GeometryData &geometry_packet);
+    std::optional<Field> getField(const SSL_GeometryData &geometry_packet) const;
```

Comment:
why do we only do optional for Field and not for the other SSL_* below?

Its clear here what happens if a `geometry_packet` is invalid, but what happens if a `detection_frame` is invalid?

Reply:
If a `detection_frame` is invalid, then it would return an empty vector